akka-stream/src/main/scala/akka/stream/OverflowStrategy.scala

```diff
@@ -3,48 +3,55 @@
  */
 package akka.stream
 
+import OverflowStrategies._
+
 /**
- * Represents a strategy that decides how to deal with a buffer that is full but is about to receive a new element.
+ * Represents a strategy that decides how to deal with a buffer of time based stage
+ * that is full but is about to receive a new element.
  */
-sealed abstract class OverflowStrategy extends Serializable
-sealed trait DelayOverflowStrategy extends Serializable
+sealed abstract class DelayOverflowStrategy extends Serializable
 
-private[akka] trait BaseOverflowStrategy {
+/**
+ * Represents a strategy that decides how to deal with a buffer that is full but is
+ * about to receive a new element.
+ */
+sealed abstract class OverflowStrategy extends DelayOverflowStrategy
 
+private[akka] object OverflowStrategies {
   /**
    * INTERNAL API
    */
-  private[akka] case object DropHead extends OverflowStrategy with DelayOverflowStrategy
-
+  private[akka] case object DropHead extends OverflowStrategy
   /**
    * INTERNAL API
    */
-  private[akka] case object DropTail extends OverflowStrategy with DelayOverflowStrategy
-
+  private[akka] case object DropTail extends OverflowStrategy
   /**
    * INTERNAL API
    */
-  private[akka] case object DropBuffer extends OverflowStrategy with DelayOverflowStrategy
-
+  private[akka] case object DropBuffer extends OverflowStrategy
   /**
    * INTERNAL API
    */
-  private[akka] case object DropNew extends OverflowStrategy with DelayOverflowStrategy
-
+  private[akka] case object DropNew extends OverflowStrategy
   /**
    * INTERNAL API
    */
-  private[akka] case object Backpressure extends OverflowStrategy with DelayOverflowStrategy
-
+  private[akka] case object Backpressure extends OverflowStrategy
   /**
    * INTERNAL API
    */
-  private[akka] case object Fail extends OverflowStrategy with DelayOverflowStrategy {
+  private[akka] case object Fail extends OverflowStrategy {
     final case class BufferOverflowException(msg: String) extends RuntimeException(msg)
```

Comment:
actually it was intentionally hidden before my this PR and I tried to preserve this approach. 
Strategies case objects are hidden but OverflowStrategy.<strategyName> are visible.
That works in case we are using stream from external code.
Still if developer wants to create own Stage that handle strategies - this will need to be public.
Let me know if we are going this way.